### PR TITLE
Anchored SnackBar on top of NoteEditor

### LINKED ISSUE #14506: [Bug] Snackbar shows in front of Note editor toolbar
###### Reproduction Steps

1. Tap the FAB to add a card to open the note editor
2. Add a card

###### Expected Result

The snackbar doesn't hide any UI content

###### Actual Result

The snackbar shows in front of the note editor toolbar

https://github.com/ankidroid/Anki-Android/assets/69634269/54cd3020-270c-4cb5-bb67-2333a9a475db

###### Debug info

2.17alpha1, Android 13

###### Research
*Enter an [x] character to confirm the points below:*

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by Giyutomioka-SS**
## Purpose / Description

Anchored the Snackbar on top of  bottom tool bar.

## Fixes 

Fixes #14506

## How Has This Been Tested?

https://github.com/ankidroid/Anki-Android/assets/91668590/25f86ec4-94ca-42fd-890f-05cdccfbbeef

Tested on Realme 6  API (30)

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Review by criticalAY — APPROVED** [review]
Looks good!

**Comment by criticalAY** [comment]
Remove the unused import

**Review by criticalAY — APPROVED** [review]
Thanks!

**Comment by BrayanDSO** [comment]
If the keyboard is hidden, the snackbar persists in the middle of the screen instead of keep being on top of the toolbar. 

@oakkitten, do you have any ideas here?

**Comment by criticalAY** [comment]
@BrayanDSO We can actually fix it by  changing it to `false` here ` android:fitsSystemWindows="true"`  I didn't notice it until I pulled the changes myself reference (here)[https://medium.com/androiddevelopers/why-would-i-want-to-fitssystemwindows-4e26d9ce1eec]

**Review by criticalAY — APPROVED** [review]
Now it seems to work fine 👍

**Review by BrayanDSO — APPROVED** [review]
LGTM. Thanks!

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt [hunk 1]
```diff
@@ -206,6 +208,10 @@ class NoteEditor : AnkiActivity(), DeckSelectionListener, SubtitleListener, Tags
             // Otherwise, display "no cards created".
         }
 
+    override val baseSnackbarBuilder: SnackbarBuilder = {
+        anchorView = findViewById<Toolbar>(R.id.editor_toolbar)
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt:212]: That controls toolbar can be hidden by the user(there's a menu option in the top toolbar menu), if it's hidden than the snackbar will appear offset with the toolbar height. The anchorView should be set only if the toolbar is set to be seen:

```
if (sharedPrefs().getBoolean(PREF_NOTE_EDITOR_SHOW_TOOLBAR, false)) {
   anchorView = findViewById<Toolbar>(R.id.editor_toolbar)
}
```

**Review by lukstbit — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt (hunk 1 above)
Comment (criticalAY) [reply AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt:212]: can be fixed by setting a condition

**Review by lukstbit — APPROVED** [review]
LGTM, thanks!

**Pull request merged**